From b490fb26ad2c38aff4d86d021701ab3ca67e0093 Mon Sep 17 00:00:00 2001
From: tomascus <arbiter1000@gmail.com>
Date: Wed, 13 Feb 2019 14:42:27 +1100
Subject: [PATCH 15/18] power: don't use SCROLL_PREFILING

The current use of the SCROLL_PREFILING hint hurts scroll performance
and results in a lot of dropped frames.

The hint ID corresponding to SCROLL_PREFILING only takes into effect for
a default timeout value of 80 ms (as found in perfboostsconfig.xml),
while the duration value in the process_interaction_hint function runs
for a much longer time, from 1500 ms up to 5000 ms (bounds of
kMinFlingDuration and kMaxInteractiveDuration). During the period
of time, when the timeout value has elapsed and current time is less
than duration, another interaction hint cannot be processed. Therefore,
if the user decides to scroll during this period, high frame drops can be
experienced as the CPU freq isn't raised by processing another hint to
account for this additional scrolling.
By only using SCROLL_VERTICAL, fluidity can be maintained, improving user
experience.

Furthmore, kMinInteractiveDuration has been lowered to 400 ms for some
platforms as it is more than sufficient to maintain fluidity (minimal
frames dropped) while scrolling.

Change-Id: I42edd07a67c7dca43a10ae2fa1b44b64b7794b43
---
 power-660.c  | 9 ++-------
 power-845.c  | 8 ++------
 power-8937.c | 8 +-------
 power-8953.c | 8 +-------
 power-8996.c | 8 +-------
 power-8998.c | 9 ++-------
 6 files changed, 9 insertions(+), 41 deletions(-)

diff --git a/power-660.c b/power-660.c
index 2df643f..a866b5b 100644
--- a/power-660.c
+++ b/power-660.c
@@ -55,8 +55,7 @@
 #define MIN_VAL(X,Y) ((X>Y)?(Y):(X))
 
 const int kMaxInteractiveDuration = 5000; /* ms */
-const int kMinInteractiveDuration = 500; /* ms */
-const int kMinFlingDuration = 1500; /* ms */
+const int kMinInteractiveDuration = 400; /* ms */
 
 static int video_encode_hint_sent;
 
@@ -283,11 +282,7 @@ static void process_interaction_hint(void *data)
     s_previous_boost_timespec = cur_boost_timespec;
     s_previous_duration = duration;
 
-    if (duration >= kMinFlingDuration) {
-        perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, -1, SCROLL_PREFILING);
-    } else {
-        perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
-    }
+    perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
 }
 
 int power_hint_override(power_hint_t hint, void *data)
diff --git a/power-845.c b/power-845.c
index 0e6a7c2..8b7fe54 100644
--- a/power-845.c
+++ b/power-845.c
@@ -59,7 +59,6 @@
 
 const int kMaxInteractiveDuration = 5000; /* ms */
 const int kMinInteractiveDuration = 100; /* ms */
-const int kMinFlingDuration = 1500; /* ms */
 
 static int display_fd;
 
@@ -236,11 +235,8 @@ static int process_interaction_hint(void *data)
     s_previous_boost_timespec = cur_boost_timespec;
     s_previous_duration = duration;
 
-    if (duration >= kMinFlingDuration) {
-        perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, -1, SCROLL_PREFILING);
-    } else {
-        perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
-    }
+    perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
+
     return HINT_HANDLED;
 }
 
diff --git a/power-8937.c b/power-8937.c
index abf0f8e..da3d0e9 100644
--- a/power-8937.c
+++ b/power-8937.c
@@ -52,7 +52,6 @@
 
 const int kMaxInteractiveDuration = 5000; /* ms */
 const int kMinInteractiveDuration = 500; /* ms */
-const int kMinFlingDuration = 1500; /* ms */
 
 static int video_encode_hint_sent;
 
@@ -228,12 +227,7 @@ static void process_interaction_hint(void *data)
     s_previous_boost_timespec = cur_boost_timespec;
     s_previous_duration = duration;
 
-    if (duration >= kMinFlingDuration) {
-        // Use launch boost resources for fling boost
-        perf_hint_enable_with_type(VENDOR_HINT_FIRST_LAUNCH_BOOST, -1, LAUNCH_BOOST_V1);
-    } else {
-        perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
-    }
+    perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
 }
 
 int power_hint_override(power_hint_t hint, void *data)
diff --git a/power-8953.c b/power-8953.c
index 06d1ead..bd9211c 100644
--- a/power-8953.c
+++ b/power-8953.c
@@ -60,7 +60,6 @@ static int display_fd;
 
 const int kMaxInteractiveDuration = 5000; /* ms */
 const int kMinInteractiveDuration = 500; /* ms */
-const int kMinFlingDuration = 1500; /* ms */
 
 static int current_power_profile = PROFILE_BALANCED;
 
@@ -326,12 +325,7 @@ static int process_interaction_hint(void *data)
     s_previous_boost_timespec = cur_boost_timespec;
     s_previous_duration = duration;
 
-    if (duration >= kMinFlingDuration) {
-        // Use launch boost resources for fling boost
-        perf_hint_enable_with_type(VENDOR_HINT_FIRST_LAUNCH_BOOST, -1, LAUNCH_BOOST_V1);
-    } else {
-        perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
-    }
+    perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
 
     return HINT_HANDLED;
 }
diff --git a/power-8996.c b/power-8996.c
index c3acab8..38ca714 100644
--- a/power-8996.c
+++ b/power-8996.c
@@ -55,7 +55,6 @@
 
 const int kMaxInteractiveDuration = 5000; /* ms */
 const int kMinInteractiveDuration = 500; /* ms */
-const int kMinFlingDuration = 1500; /* ms */
 
 static int current_power_profile = PROFILE_BALANCED;
 
@@ -312,12 +311,7 @@ static int process_interaction_hint(void *data)
     s_previous_boost_timespec = cur_boost_timespec;
     s_previous_duration = duration;
 
-    if (duration >= kMinFlingDuration) {
-        // Use launch boost resources for fling boost
-        perf_hint_enable_with_type(VENDOR_HINT_FIRST_LAUNCH_BOOST, -1, LAUNCH_BOOST_V1);
-    } else {
-        perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
-    }
+    perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
 
     return HINT_HANDLED;
 }
diff --git a/power-8998.c b/power-8998.c
index 20804ab..123c179 100644
--- a/power-8998.c
+++ b/power-8998.c
@@ -55,8 +55,7 @@
 #define NUM_PERF_MODES  3
 
 const int kMaxInteractiveDuration = 5000; /* ms */
-const int kMinInteractiveDuration = 500; /* ms */
-const int kMinFlingDuration = 1500; /* ms */
+const int kMinInteractiveDuration = 400; /* ms */
 
 static int current_power_profile = PROFILE_BALANCED;
 
@@ -308,11 +307,7 @@ static int process_interaction_hint(void *data)
     s_previous_boost_timespec = cur_boost_timespec;
     s_previous_duration = duration;
 
-    if (duration >= kMinFlingDuration) {
-        perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, -1, SCROLL_PREFILING);
-    } else {
-        perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
-    }
+    perf_hint_enable_with_type(VENDOR_HINT_SCROLL_BOOST, duration, SCROLL_VERTICAL);
 
     return HINT_HANDLED;
 }
-- 
2.17.1

